### DISCUSSION TITLE
emqx http publish api time cost too long (over 300ms)

### DISCUSSION THREAD

**Discussion by wangfeiing** [emqx/emqx#12003] [Q&A]
URL：http://{EMXQ-IP}:8080/api/v5/publish
We made 482 http requests, and 8 of them timed out with a timeout of 300 milliseconds. 

So I would like to seek help from the official  about whether the HTTP API request time consumption meets expectations.

And is there an optimization method to shorten the time spent on HTTP API requests？:)  Thanks

**Comment by ieQu1** [comment]
Which EMQX version are you using? What was the payload size? What kind of hardware EMQX is running on (number of cores, RAM)? Did any message appear in emqx.log?
